src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering.LocalRewriter
+{
+    class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuard
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return (BoundNode)localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            int current = 0;
+            ImmutableArray<BoundStatement> statements = (ImmutableArray<BoundStatement>)this.VisitList(node.Statements);
+
+            foreach (BoundStatement statement in statements)
+            {
+                if (statement is BoundLocalDeclaration boundAssignment)
+                {
+                    if (boundAssignment.LocalSymbol.IsUsing)
+                    {
+                        ImmutableArray<LocalSymbol> locals = ImmutableArray.Create<LocalSymbol>(boundAssignment.LocalSymbol);
+
+                        List<BoundStatement> precedingStatements = new List<BoundStatement>();
```

Comment:
>List [](start = 71, length = 4)

Can we use a pooled ArrayBuilder instead?

Reply:
In addition to the pool it seems like we can easily pass a capacity argument to the pool as we know the number of elements being added. 

---
In reply to: [202866335](https://github.com/dotnet/roslyn/pull/28587#discussion_r202866335) [](ancestors = 202866335)